astrology.lic

```diff
@@ -90,56 +171,72 @@ class Astrology
                    result = nil
                    until ['Roundtime', 'You peer aimlessly through your telescope'].include?(result)
                      if Flags['bad-search'] == 'is foiled by the daylight'
-                       train_astrology
+                       check_heavens
                        return
                      elsif Flags['bad-search'] == 'turns up fruitless'
                        next
                      end
-                     result = bput('peer telescope', 'You peer aimlessly through your telescope', 'You see nothing regarding the future', 'Roundtime')
+                     result = observe(data['name'])
                    end
                    result == 'Roundtime'
                  end
                else
-                 until ['Roundtime', 'Clouds obscure'].include? bput("observe #{best_eye_data['name']} in sky", 'Your search for', 'You see nothing regarding the future', 'Clouds obscure', 'Roundtime')
+                 until observe(best_eye_data['name'])
                    if Flags['bad-search']
-                     train_astrology
+                     check_heavens
                      return
                    end
                  end
                  best_eye_data
-               end
+          end
+
     pause 2
     waitrt?
 
-    bput('stow telescope', 'you put') if @have_telescope
-
     observed['pools']
       .reject { |_skill, value| value.nil? }
       .each { |skill, _value| align(skill) }
-    bput('predict anal', 'roundtime')
     waitrt?
   end
 
+  def observe(body)
+    Flags.add('bad-search', 'is foiled by the (daylight|darkness)', 'turns up fruitless')
+
+    if @have_telescope
+      bput('get telescope', 'you get', 'You are already', "That can't be picked up")
+      bput("center telescope on #{body}", 'You put your eye')
+      result = bput('peer telescope', 'You peer aimlessly through your telescope', 'You see nothing regarding the future', 'Roundtime')
+      bput('stow telescope', 'you put', "That can't be picked up")
+      result
+    else
+      ['Roundtime', 'Clouds obscure'].include? bput("observe #{body} in sky", 'Your search for', 'You see nothing regarding the future', 'Clouds obscure', 'Roundtime')
+    end
+  end
+  
   def roll_bones
     if @divination_bones_tied
       bput("untie bones from my #{@divination_bones_container}", 'divination bones')
     else
-      bput("get bones from my #{@divination_bones_container}", 'divination bones')
+      bput("get bones from my #{@divination_bones_container}", 'divination bones', 'you get')
     end
     bput('roll my bones', 'roundtime')
     waitrt?
     if @divination_bones_tied
       bput("tie bones to my #{@divination_bones_container}", 'divination bones')
     else
-      bput("put bones in my #{@divination_bones_container}", 'divination bones')
+      bput("put bones in my #{@divination_bones_container}", 'divination bones', 'you put')
```

Comment:
I'm guessing the person who added the bones usage has a different set than I do. My getting/stowing commands were not responding to 'divination bones', so I added the general text for getting or storing something.